### PR TITLE
Move azure storage resource to bicep model

### PR DESCRIPTION
**Pull request by davidfowl**
- Changed blobs, tables and queues to be "value.v0" since they don't have bicep files.
- Remove storage provisioner
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/2329)

### CODE REVIEW

File: playground/bicep/BicepSample.AppHost/Program.cs [hunk 1]
```diff
@@ -15,12 +15,12 @@
...
 var kv = builder.AddAzureKeyVault("kv");
 var appConfig = builder.AddAzureAppConfiguration("appConfig").WithParameter("sku", "standard");
-var storage = builder.AddAzureBicepAzureStorage("storage");
+var storage = builder.AddAzureStorage("storage");
                     // .UseEmulator();
 
-var blobs = storage.AddBlob("blob");
-var tables = storage.AddTable("table");
-var queues = storage.AddQueue("queue");
+var blobs = storage.AddBlobs("blob");
```
Comment (mitchdenny) [inline playground/bicep/BicepSample.AppHost/Program.cs:21]: Worth keeping the old methods with obsolete tag?

File: src/Aspire.Hosting.Azure/AzureQueueStorageResource.cs [hunk 2]
```diff
@@ -13,13 +16,29 @@ public class AzureQueueStorageResource(string name, AzureStorageResource storage
     IResourceWithParent<AzureStorageResource>
 {
     /// <summary>
-    /// Gets the parent AzureStorageResource of this AzureQueueStorageResource.
+    /// Gets the parent AzureStorageResource of this AzureBlobStorageResource.
```
Comment (mitchdenny) [inline src/Aspire.Hosting.Azure/AzureQueueStorageResource.cs:19]: Type incorrect?

File: src/Aspire.Hosting.Azure/AzureQueueStorageResource.cs [hunk 3]
```diff
@@ -13,13 +16,29 @@ public class AzureQueueStorageResource(string name, AzureStorageResource storage
...
     /// </summary>
     public AzureStorageResource Parent => storage;
 
     /// <summary>
-    /// Gets the connection string for the Azure Queue Storage resource.
+    /// Gets the connection string template for the manifest for the Azure Blob Storage resource.
```
Comment (mitchdenny) [inline src/Aspire.Hosting.Azure/AzureQueueStorageResource.cs:24]: Copypasta?

File: src/Aspire.Hosting.Azure/AzureTableStorageResource.cs [hunk 4]
```diff
@@ -13,13 +16,29 @@ public class AzureTableStorageResource(string name, AzureStorageResource storage
     IResourceWithParent<AzureStorageResource>
 {
     /// <summary>
-    /// Gets the parent AzureStorageResource of this AzureTableStorageResource.
+    /// Gets the parent AzureStorageResource of this AzureBlobStorageResource.
```
Comment (mitchdenny) [inline src/Aspire.Hosting.Azure/AzureTableStorageResource.cs:19]: Copy pasta error?

File: src/Aspire.Hosting.Azure/AzureTableStorageResource.cs [hunk 5]
```diff
@@ -13,13 +16,29 @@ public class AzureTableStorageResource(string name, AzureStorageResource storage
...
     /// </summary>
     public AzureStorageResource Parent => storage;
 
     /// <summary>
-    /// Gets the connection string for the Azure Table Storage resource.
+    /// Gets the connection string template for the manifest for the Azure Blob Storage resource.
```
Comment (mitchdenny) [inline src/Aspire.Hosting.Azure/AzureTableStorageResource.cs:24]: Copypasta

File: src/Aspire.Hosting.Azure/AzureTableStorageResource.cs [hunk 6]
```diff
@@ -13,13 +16,29 @@ public class AzureTableStorageResource(string name, AzureStorageResource storage
...
+    /// </summary>
+    public string ConnectionStringExpression => Parent.TableEndpoint.ValueExpression;
+
+    /// <summary>
+    /// Gets the connection string for the Azure Blob Storage resource.
     /// </summary>
-    /// <returns>The connection string for the Azure Table Storage resource.</returns>
+    /// <returns>The connection string for the Azure Blob Storage resource.</returns>
```
Comment (mitchdenny) [inline src/Aspire.Hosting.Azure/AzureTableStorageResource.cs:31]: Copypasta

File: src/Aspire.Hosting.Azure/AzureTableStorageResource.cs [hunk 7]
```diff
@@ -13,13 +16,29 @@ public class AzureTableStorageResource(string name, AzureStorageResource storage
...
     public string? GetConnectionString() => Parent.GetTableConnectionString();
+
+    /// <summary>
+    /// TODO: Doc Comments
```
Comment (mitchdenny) [inline src/Aspire.Hosting.Azure/AzureTableStorageResource.cs:35]: Doc comments

File: tests/Aspire.Hosting.Tests/ManifestGenerationTests.cs [hunk 8]
```diff
@@ -328,37 +328,6 @@ public void EnsureAllPostgresManifestTypesHaveVersion0Suffix()
...
-
-        program.Run();
-
-        var resources = publisher.ManifestDocument.RootElement.GetProperty("resources");
-
-        var storage = resources.GetProperty("storage");
-        Assert.Equal("azure.storage.v0", storage.GetProperty("type").GetString());
-
-        var blobs = resources.GetProperty("blobs");
-        Assert.Equal("azure.storage.blob.v0", blobs.GetProperty("type").GetString());
```
Comment (mitchdenny) [inline tests/Aspire.Hosting.Tests/ManifestGenerationTests.cs:353]: These tests will probably go away eventually ... but for now just change them to value.v0 for the sub-resources?

**Review by mitchdenny — APPROVED** [review]
LGTM sans a few doc comment/copypasta issues.

File: playground/bicep/BicepSample.AppHost/aspire-manifest.json [hunk 9]
```diff
@@ -51,17 +51,17 @@
       }
     },
     "blob": {
-      "type": "azure.bicep.v0",
+      "type": "value.v0",
       "connectionString": "{storage.outputs.blobEndpoint}",
       "parent": "storage"
```
[the remark this answers is not in the document]
Comment (davidfowl) [reply playground/bicep/BicepSample.AppHost/aspire-manifest.json:56]: Yea lets. I can do it here.

File: src/Aspire.Hosting.Azure/AzureTableStorageResource.cs (hunk 4 above)
Comment (davidfowl) [reply src/Aspire.Hosting.Azure/AzureTableStorageResource.cs:19]: hmmm

File: playground/bicep/BicepSample.AppHost/Program.cs (hunk 1 above)
Comment (davidfowl) [reply playground/bicep/BicepSample.AppHost/Program.cs:21]: nah this one feels too easy to fix.

File: tests/Aspire.Hosting.Tests/ManifestGenerationTests.cs (hunk 8 above)
Comment (davidfowl) [reply tests/Aspire.Hosting.Tests/ManifestGenerationTests.cs:353]: I have replacement tests for this alraedy.

**Pull request merged**